File: Online_Health_Consultation_Portal/Application/Dtos/Users/UserWithProfile.cs
using Online_Health_Consultation_Portal.Domain;

namespace Online_Health_Consultation_Portal.Application.Dtos.Users
{
    public class UserWithProfile
    {
        public User User { get; set; }
        public Doctor? Doctor { get; set; }
        public Patient? Patient { get; set; }
    }
}

File: FE_eHealth/source/main/src/app/core/models/userWithProfile.ts
import { User } from "@core/service/user.service";
import { Doctors } from "app/admin/doctors/alldoctors/doctors.model";
import { PatientProfile } from "app/admin/patients/allpatients/patient.model";

export interface UserWithProfile {
  user: User;
  doctor?: Doctors | null;
  patient?: PatientProfile | null;
}
